images/win/scripts/Installers/Install-Msys2.ps1

```diff
@@ -0,0 +1,76 @@
+################################################################################
+##  File:  Install-Msys2.ps1
+##  Desc:  Install Msys2 and 64-bit gcc, cmake, & llvm (32-bit commented out)
+################################################################################
+
+# References
+# https://github.com/msys2/MINGW-packages/blob/master/azure-pipelines.yml
+# https://packages.msys2.org/group/
+
+$origPath = $env:PATH
+$gitPath  = "$env:ProgramFiles\Git"
+
+# get info from https://sourceforge.net/projects/msys2/files/Base/x86_64/
+$msys2Uri  = "http://repo.msys2.org/distrib/x86_64/msys2-base-x86_64-20190524.tar.xz"
+$msys2File = "$env:TEMP\msys2.tar.xz"
+
+# Download the latest msys2 x86_64
+Write-Host "Starting msys2 download"
+(New-Object System.Net.WebClient).DownloadFile($msys2Uri, $msys2File)
+Write-Host "Finished download"
+
+$msys2FileU = "/$msys2File".replace(':', '')
+
+$tar = "$gitPath\usr\bin\tar.exe"
+
+# extract tar.xz to C:\
+Write-Host "Starting msys2 extraction"
+&$tar -Jxf $msys2FileU -C /c/
+Remove-Item $msys2File
+Write-Host "Finished extraction"
+
+# Add msys2 bin tools folders to PATH temporary
+$env:PATH = "C:\msys64\mingw64\bin;C:\msys64\usr\bin;$origPath"
+
+Write-Host "bash pacman-key --init"
+bash.exe -c "pacman-key --init 2>&1"
+
+Write-Host "bash pacman-key --populate msys2"
+bash.exe -c "pacman-key --populate msys2 2>&1"
+
+Write-Host "pacman --noconfirm -Syyuu"
```

Comment:
Ref https://github.com/actions/virtual-environments/pull/342#issuecomment-581086902

Reply:
This PR is based on @MSP-Greg suggested one that's why it's here. I'd prefer to wait for image deployment and then we can collect some feedback about working scenarios and decide if this approach suits or not.

Reply:
Actually, I'm good with the merged solution, even though it constraints the possible use cases. As commented in previous discussions, users can keep using other custom MSYS2 installations through action setup-msys2. I'd just like it to be explicitly documented, so that users of GitHub Actions are aware that this is not a default MSYS2 install.

Reply:
Could you provide your vision on documentation, please? What points should be mentioned?

Reply:
Sure. For example:

> In order to reduce startup time for most common use cases, the MSYS2 installation provided in `windows` environments is installed from binaries, then updated, and some packages such as `base-devel`, `compression`, `clang` and `cmake` are preinstalled. Please open an issue in LINK for requesting additional packages to be preinstalled.
>
> As a result, users willing to use a stable environment should install MSYS2 themselves. See actions/virtual-environments#632 for a reference. Alternatively Action setup-msys2 can be used.

Reply:
Hi @eine, we will add information about non-default MSYS2 installation to docs, thank you for your participate!